src/Engine/ProtoCore/AssociativeGraph.cs

```diff
@@ -524,7 +497,7 @@ public static AssociativeGraph.GraphNode MarkGraphNodeDirty(RuntimeCore runtimeC
         /// <param name="core"></param>
         /// <param name="nodeList"></param>
         /// <returns></returns>
-        public static AssociativeGraph.GraphNode MarkGraphNodesDirty(Core core, IEnumerable<AST.AssociativeAST.AssociativeNode> nodeList)
+        public static AssociativeGraph.GraphNode MarkGraphNodesDirty(RuntimeCore core, IEnumerable<AST.AssociativeAST.AssociativeNode> nodeList)
```

Comment:
Can you rename the function from 'MarkGraphNodesDirty' to 'MarkGraphNodesDirtyAtGlobalScope'